File: api/DropShot.Application/Products/Models/ProductDetailsDto.cs
using DropShot.Application.Common.AutoMapper;
using DropShot.Domain.Entities;
using DropShot.Domain.Enums;

namespace DropShot.Application.Products.Models;

public class ProductDetailsDto : IMapFrom<Product>
{
    public int Id { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public decimal Price { get; set; }
    public ClothesUnitOfMeasure UnitOfSize { get; set; }
    public ICollection<VariantOnProductDto> Variants { get; set; }
}

File: web/src/api/models/Products/ProductDetailsDto.ts
import { ClothesUnitOfMeasure } from './ClothesUnitOfMeasure';
import { VariantOnProductDto } from './VariantOnProductDto';

export interface ProductDetailsDto {
  id: number;
  name: string;
  description: string;
  price: number;
  unitOfSize: ClothesUnitOfMeasure;
  variants: VariantOnProductDto[];
}
